### DISCUSSION TITLE
General understanding question: How to handle the CRUD-like update in event sourcing?

### DISCUSSION THREAD

**Discussion by ahoiroman** [spatie/laravel-event-sourcing#358] [Q&A]
Hello there,

at the moment, I am digging deeper in event sourcing. I used CRUD for a long time now and I guess I'm still kinda stuck in the CRUD-way.

Well, now to my question:

I event-sourced a part of my application, where I create something called a `Job`. A `Job` can have:

- `title`
- `description`
- `created_at`

So creating this `Job` is easy - but what do I do when it comes to updating?

Is it an anti pattern to dispatch an event like `JobUpdated`, which contains changes to the `title` and possibly the `description`? Or should I dispatch multiple events like:

 - `JobTitleChanged`
 - `JobDescriptionChanged`

**Comment by maartenpaauw** [comment]
Just curious, what decision did you make in the end: a `JobUpdated` event or a event for each attribute (.e.g. `JobTitleChanged`) change?

**Comment by sebastiandedeyne** [comment]
I just wrote a post on this: https://sebastiandedeyne.com/granular-events

Tl;dr: If the data is not critical or most importantly you don't plan to react to, large events are fine. Otherwise, granular events are the more versatile choice.
